drivers/wifi/esp/esp.h

```diff
@@ -95,6 +95,25 @@ extern "C" {
 #define ESP_CONNECT_TIMEOUT	K_SECONDS(20)
 #define ESP_INIT_TIMEOUT	K_SECONDS(10)
 
+
+#define ESP_MODE_STATION	1
+#if defined(CONFIG_WIFI_ESP_AT_VERSION_1_7)
+#define ESP_MODE_SOFTAP		0
+#else
+#define ESP_MODE_SOFTAP		2
+#endif
```

Comment:
I've looked into docs if those esp mode values are shared across different commands. Unfortunately that is not true for `CONFIG_WIFI_ESP_AT_VERSION_1_7` case, as `AT+CWMODE_CUR` assumes SoftAP mode to be represented by value `2`, but for `AT+CWMODE_CUR` command by value `0` (as you've implemented above).

So either we need those `ESP_MODE_*` macros to be renamed to `ESP_CWDHCP_MODE_*` or somehow map `ESP_MODE_*` to resolve to different values depending on command in which they are used. The former approach seems easier, but more verbose. For now I don't know how to implement the latter...

Reply:
> Some few more things in comments. Additionally I don't think `- Add ESP DHCP Support` is the correct sentence in git commit message, as well as `esp DHCP/Static IP Support` in commit title. We already relied on DHCP to be working. The only thing that changes now is that you force DHCP by calling `AT+CWDHCP` command, instead of relying it to be the default in ESP firmware.

On firmware v1.7.4 this never worked for me on esp8266-01, I tried also to do `AT+RESTORE` then connect, but either ESP fails to get an IP Address and disconnect or it got some weird IP starts with `252.xxx.xxx.xxx`

Also, I think it is better to get things under control from Zephyr side, so we make sure that our AT commands sequence are working and we don't rely on ESP init sequence which might change.

Reply:
Good catch ! 👍
Will think about it

Reply:
Solved as agreed on Slack
Thanks for your support.